packages/lab/src/custom-slider/CustomSlider.tsx

```diff
@@ -0,0 +1,208 @@
+   */
+  step?: number;
+  /**
+   * Initial value of the slider
+   */
+  defaultValue?: number;
+  /**
+   * Position of the labels
+   */
+  labelPosition?: "bottom" | "inline";
+  /**
+   * Show bottom markers
+   */
+  showMarkers?: boolean;
+  /**
+   * Value of the slider, to be used when in a controlled state
+   */
+  value?: number;
+  /**
+   * Change handler to be used when in a controlled state
+   */
+  onChange?: (value: number) => void;
+}
+
+export const CustomSlider = forwardRef<HTMLDivElement, SliderProps>(
+  function Slider(
+    {
+      min = 0,
+      max = 10,
+      step = 1,
+      value: valueProp,
+      defaultValue = 0,
+      onChange,
+      className,
+      "aria-label": ariaLabel,
+      "aria-valuetext": ariaValueText,
+      labelPosition = "inline",
+      showMarkers = false,
+      ...rest
+    },
+    ref,
+  ) {
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-slider",
+      css: sliderCss,
+      window: targetWindow,
+    });
+
+    const [value, setValue] = useControlled({
+      controlled:
+        valueProp === undefined ? undefined : clamp(valueProp, min, max),
+      default: defaultValue,
+      name: "Slider",
+      state: "value",
+    });
+
+    const [tooltipVisible, setTooltipVisible] = useState(false);
+    const [isDragging, setIsDragging] = useState(false);
+    const sliderRef = useRef<HTMLDivElement>(null);
+
+    const calculateThumbPosition = (clientX: number) => {
+      if (sliderRef.current) {
+        const sliderRect = sliderRef.current.getBoundingClientRect();
+        const rawValue =
+          ((clientX - sliderRect.left) / sliderRect.width) * (max - min) + min;
+        const steppedValue = Math.round(rawValue / step) * step;
+        const newValue = Math.min(Math.max(min, steppedValue), max);
+        setValue(newValue);
+        onChange?.(newValue);
+      }
+    };
+
+    const handleMouseMove = (event: MouseEvent) => {
+      calculateThumbPosition(event.clientX);
+    };
+
+    const handleMouseUp = () => {
+      setIsDragging(false);
+      setTooltipVisible(false);
+      document.removeEventListener("mousemove", handleMouseMove);
+      document.removeEventListener("mouseup", handleMouseUp);
+    };
+
+    const handleMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+      calculateThumbPosition(event.clientX);
+      setIsDragging(true);
+      document.addEventListener("mousemove", handleMouseMove);
+      document.addEventListener("mouseup", handleMouseUp);
+    };
+
+    const handleMouseEnter = () => {
+      setTooltipVisible(true);
+    };
+
+    const handleMouseLeave = () => {
+      if (!isDragging) setTooltipVisible(false);
+    };
+
+    const handleFocus = () => setTooltipVisible(true);
+
+    const handleBlur = () => setTooltipVisible(false);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+      let newValue = value;
+      switch (event.key) {
```

Comment:
We should align with StepperInput's keyboard handling i.e. PageUp or Shift + ArrowUp